src/RectangleB.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Maman 12 - Class RectangleB.
+ * The RectangleB class represents a 2D axis aligned integral rectangles.
+ *
+ * @author Noa Amir
+ * @version 2022a
+ */
+public class RectangleB {
+    // 1. instance variables
+    private Point _pointSW;
+    private Point _pointNE;
+
+    // 2. constructors
+
+    /**
+     * First constructor for objects of class RectangleB Constructs a new rectangle with the specified width, height and it's south west corner is (0,0)
+     *
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleB(int w, int h) {
+        _pointSW = new Point(0, 0);
+
+        if (w < 1)
+            w = 1;  // TODO: Should I do final DEFAULT_WIDTH = 1
+
+        if (h < 1)
+            h = 1; // TODO: Should I do final DEFAULT_HEIGHT = 1
+
+        _pointNE = new Point(w, h);
+    }
+
+    /**
+     * Second constructor for objects of class RectangleB Constructs a new rectangle with the specified vertices
+     *
+     * @param p - south-western vertex
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleB(Point p, int w, int h) {
+        _pointSW = new Point(p);
+
+        if (w < 1)
+            w = 1;
+
+        if (h < 1)
+            h = 1;
+
+        _pointNE = new Point(_pointSW.getX() + w, _pointSW.getY() + h);
+    }
+
+    /**
+     * Third constructor for objects of class RectangleB Constructs a new rectangle with the specified vertices
+     *
+     * @param sw - south-western vertex
+     * @param ne - north-eastern vertex
+     */
+    public RectangleB(Point sw, Point ne) {
+        _pointSW = new Point(sw);
+        _pointNE = new Point(ne);
+    }
+
+    /**
+     * Copy constructor for objects of class RectangleB Constructs a rectangle using another rectangle
+     *
+     * @param r - The rectangle from which to construct the new object
+     */
+    public RectangleB(RectangleB r) {
+        _pointSW = new Point(r._pointSW);
+        _pointNE = new Point(r._pointNE);
+    }
+
+    // 3. getters and setters
+
+    /**
+     * Returns the South-West point of the rectangle
+     *
+     * @return a copy of the s-w point of the rectangle
+     */
+    public Point getPointSW() {
+        return new Point(_pointSW);
+    }
+
+    /**
+     * Returns the North-East point of the rectangle
+     *
+     * @return a copy of the n-e point of the rectangle
+     */
+    public Point getPointNE() {
+        return new Point(_pointNE);
+    }
+
+    /**
+     * Sets the South-West point of the rectangle.
+     * This change affects also the NE Point since the width and height of the rectangle should NOT change
+     *
+     * @param p - the new S-W point of the rectangle
+     */
+    // TODO: NOT SURE!
+    public void setPointSW(Point p) {
+        int deltaX = p.getX() - _pointSW.getX();
+        int deltaY = p.getY() - _pointSW.getY();
+        //_pointSW = new Point(p);
```

Comment:
leftover?

Reply:
Yes, debating how to do it. Few ways here.
The moving looks ok?